tidy3d/components/simulation.py

```diff
@@ -3618,6 +3618,53 @@ def _source_homogeneous_isotropic(cls, val, values):
                                 "A fixed angle plane wave can only be injected into a homogeneous isotropic"
                                 "dispersionless medium."
                             )
+                    # check if broadband angled gaussian beam frequency variation is too fast
+                    if (
+                        isinstance(source, (GaussianBeam, AstigmaticGaussianBeam))
+                        and np.abs(source.angle_theta) > 0
+                        and source.num_freqs > 1
+                    ):
+
+                        def radius(waist_radius, waist_distance, k0):
+                            """Gaussian beam radius at a given waist distance and k0."""
+                            z_r = waist_radius**2 * k0 / 2
+                            return waist_radius * np.sqrt(1 + (waist_distance / z_r) ** 2)
+
+                        # A slanted GaussianBeam will accumulate a phase that's frequency-dependent
+                        # like phi = K f, with the derivative dphi / df = K = 2 * pi * n * r * sin(theta) / c_0.
+                        # Here, we compute the maximum value of this coefficient computed at the waist radius
+                        # and over all frequencies. Then we compare this to the frequency spacing to
+                        # determine whether the frequency dependence is too fast, and issue a warning.
+                        dph_df = []
```

Comment:
a totally optional suggestion to improve readability: 
instead of `dph_df` we can compute `optical_path_length.append(n * r * sin(theta))`, then later 
```
max_path_length = np.max(optical_path_length)
max_phase_difference = max_path_length * 2 * pi * (freqs[-1] - freqs[0]) / c_0
if not max_phase_difference / source.num_freqs < 1:
    ...
```

Reply:
Sure, why not. :)